src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -1667,17 +1698,17 @@ public void Warning(string p, bool isPersistent = false)
             if (isPersistent)
             {
                 State = ElementState.PersistentWarning;
-                if (!string.Equals(persistentWarning, p))
+                if (!persistentWarnings.Contains(p))
```

Comment:
Since `persistentWarnings` is a hashset, this check may not be required, right?

Reply:
uhmm..I think you are right about that 
THanks